### DISCUSSION TITLE
Modify nginx config

### DISCUSSION THREAD

**Discussion by radokristof** [nfrastack/container-lemonldap#43] [Q&A] [answered]
What is the recommended way to alter the nginx configuration? Copy contents & map it to a folder?

As per the documentation, I need to add a rewrite rule for Mattermost: https://lemonldap-ng.org/documentation/2.0/applications/mattermost.html

**Comment by tiredofit** [comment]
There are a couple ways of doing this, rather than building a whole new image the easiest, most portable approach would be to utilize the `/assets/custom-scripts` folder. This will execute upon startup a bash script (in alphabetical order, starting with numbers ie 00-script.sh, 00-temp-script.sh, 01-script.sh) and execute whatever you'd like on the internal container file system.

With the bash script you could pretty much do whatever you'd like to any of the internal configuration  with `cat`, `sed`, or `awk`. In this case you'd want to insert the contents into `/etc/nginx/sites.available/portal.conf`

Note, if you use `#!/command/with-contenv bash` as a first line of your script it will read your container environment variables which may be of use.

**Reply by tiredofit** [reply, marked as answer]
See https://github.com/tiredofit/docker-lemonldap/commit/9dd95e7e87b9a3c5cdbca9ddea4071e283a0e4f2

It's available with `2.0.66` and later..

**Reply by radokristof** [reply]
Thank you for the updated image and also the idea on how to manage custom container assets. It also gave an idea on how to effectively mount & copy skins, logos, etc... Thank you :)

**Reply by tiredofit** [reply]
I have an even easier method for that :) Undocumented of course.

Try dropping your files in `/assets/custom` and upon bootup they will get copied into /usr/share/lemonldap-ng and either overwrite the existing files and folder structure or append to it. It will also change the ownership permissions to be read by the webserver for specifically this purpose of introducing logos/themes..

https://github.com/tiredofit/docker-lemonldap/blob/d990670eca0f6f3c8d781a509afd3a19cdae4ee9/install/etc/cont-init.d/20-llng#L35

**Reply by radokristof** [reply]
Thank you very much, I will test this :)
